torch/csrc/api/include/torch/data/samplers/base.h

```diff
@@ -15,37 +15,29 @@ class InputArchive;
 namespace torch {
 namespace data {
 namespace samplers {
-
-/// A `Sampler` is an object that yields indices with which to index into a
+/// A `Sampler` is an object that yields an index with which to access a
 /// dataset.
+template <typename Index = std::vector<size_t>>
 class Sampler {
  public:
+  using IndexType = Index;
```

Comment:
Why can't you simply use `Index` here?

Reply:
These typedefs are to allow fetching the type later through `typename Sampler::IndexType`

Reply:
Sure I understand their purpose. The problem is that you sometimes call those type aliases `X` and sometimes `XType` which is quite inconsistent.